src/include/executor/hashjoin.h

```diff
@@ -286,6 +286,7 @@ typedef struct ParallelHashJoinState
 	Barrier		grow_buckets_barrier;
 	Barrier		sync_barrier;
 	Barrier		batch0_barrier;
+	volatile 	bool	phs_lasj_has_null; 	/* LASJ has found null value, identify early quit */
```

Comment:
should we use `pg_atomic_flag` to avoid barrier usage ?

Reply:
Sounds good, I will take a look at it.

Reply:
> should we use pg_atomic_flag to avoid barrier usage ?

@yjhjstz 
I dig into for a minute, `pg_atomic_flag` is used for atomic value for more efficiency on non-Intel platforms(is used for x86[_64]).
And it has nothing to do with barrier, we still need barrier to sync between CPU/cache if use pg_atomic_flag.
As phs_lasj_has_null is a simple bool, i think we don't need to introduce a struct with more complexity.